### PR TITLE
fix: fix error handling in the deployments/next handler

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
PR includes a fix for error handling in the deployments/next handler, but it also unifies device deployment request model, between OS and Enterprise versions of the service (https://tracker.mender.io/browse/MEN-5597)

### CODE REVIEW

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

File: model/device_deployment.go [hunk 1]
```diff
@@ -353,8 +354,31 @@ func InactiveDeploymentStatuses() []DeviceDeploymentStatus {
...
+func (i *DeploymentNextRequest) Validate() error {
+	return validation.ValidateStruct(i,
+		validation.Field(&i.DeviceProvides,
+			validation.Required,
+		),
+	)
+}
+
+func (i *DeploymentNextRequest) String() string {
+	json, err := json.Marshal(i)
```
Comment (alfrunes) [inline model/device_deployment.go:377]: nit(lint): Variable overriding `json` package name.

File: model/device_deployment.go [hunk 2]
```diff
@@ -353,8 +354,31 @@ func InactiveDeploymentStatuses() []DeviceDeploymentStatus {
...
+
+func (i *DeploymentNextRequest) Validate() error {
+	return validation.ValidateStruct(i,
+		validation.Field(&i.DeviceProvides,
+			validation.Required,
+		),
+	)
+}
+
+func (i *DeploymentNextRequest) String() string {
```
Comment (alfrunes) [inline model/device_deployment.go:376]: Q: What do we need the `fmt.Stringer` interface for?

File: store/mongo/datastore_mongo.go [hunk 3]
```diff
@@ -1306,13 +1305,36 @@ func (db *DataStoreMongo) UpdateDeviceDeploymentLogAvailability(ctx context.Cont
...
+	request *model.DeploymentNextRequest,
+) error {
+
+	database := db.client.Database(mstore.DbFromContext(ctx, DatabaseName))
+	collDevs := database.Collection(CollectionDevices)
+
+	res, err := collDevs.UpdateOne(
+		ctx,
+		bson.D{{Key: StorageKeyId, Value: ID}},
+		bson.D{{Key: "$set", Value: bson.M{StorageKeyDeviceDeploymentRequest: request}}},
```
Comment (alfrunes) [inline store/mongo/datastore_mongo.go:1322]: Could there be instances where the device deployment document does not (yet) exist?

**Review by alfrunes — APPROVED** [review]
The changes look very familiar.
I put in a couple of comments just to give some additional input on a few observations.
Feel free to ignore.

File: model/device_deployment.go (hunk 1 above)
Comment (kjaskiewiczz) [reply model/device_deployment.go]: fixed

File: model/device_deployment.go (hunk 2 above)
Comment (kjaskiewiczz) [reply model/device_deployment.go:376]: only to nicely log old and new request when they are different

File: store/mongo/datastore_mongo.go (hunk 3 above)
Comment (kjaskiewiczz) [reply store/mongo/datastore_mongo.go:1322]: to be able to save the request you have to have the device deployment ID, so I think the object should be always there

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline

**Review by tranchitella — APPROVED** [review]
Looks good. My only comment is about the commit message:

"chore: fix error handling in the deployments/next handler"

Is it maintenance (chore) or a fix? :)
I'd either use

"chore: improve error handling in the deployments/next handler"

or

"fix: error handling in the deployments/next handler"

**Comment by kjaskiewiczz** [comment]
pipeline: https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/524191801

**Pull request merged**